src/Spice86.Core/Emulator/CPU/CfgCpu/ParsedInstruction/Discriminator.cs

```diff
@@ -0,0 +1,130 @@
+namespace Spice86.Core.Emulator.CPU.CfgCpu.ParsedInstruction;
+
+using System.Collections.Immutable;
+
+/// <summary>
+/// Base class for objects that have a discriminator value based on a list of bytes.
+/// The bytes are used to compare 2 discriminators.
+/// If at the same position bytes are equal, the discriminators are considered equals.
+/// If there are null bytes they are not considered for the equality comparison of the position
+/// If the length of the discriminator value differs, discriminators are different
+/// </summary>
+public class Discriminator : IComparable<Discriminator> {
+    public Discriminator(ImmutableList<byte?> discriminatorValue) {
+        DiscriminatorValue = discriminatorValue;
+    }
+
+    /// <summary>
+    /// Value of the discriminator
+    /// </summary>
+    public ImmutableList<byte?> DiscriminatorValue { get; }
+
+    /// <inheritdoc/>
+    public int CompareTo(Discriminator? other) {
+        if (other == null) {
+            return 1;
+        }
+
+        int count = DiscriminatorValue.Count;
+        if (count != other.DiscriminatorValue.Count) {
+            return count.CompareTo(other.DiscriminatorValue.Count);
+        }
+        // Size is equal, let's compare the elements of the discriminators
+        for (int i = 0; i < count; i++) {
+            byte? thisByte = DiscriminatorValue[i];
+            byte? otherByte = other.DiscriminatorValue[i];
+            if (thisByte != otherByte) {
+                // null is equals here
+                if (thisByte == null || otherByte == null) {
+                    continue;
+                }
+                
+                return thisByte.Value.CompareTo(otherByte.Value);
+            }
+        }
+        // Equals
+        return 0;
+    }
+    
+    /// <summary>
+    /// Checks that the given span of bytes is equivalent to the discriminator.
+    /// Equivalence means that they have the same length and their content is identical at each position or null
+    /// </summary>
+    /// <param name="bytes">Span of bytes to compare with the discriminator</param>
+    /// <returns>true if they are equivalent, false otherwise</returns>
+    public bool SpanEquivalent(Span<byte> bytes) {
```

Comment:
This could use Span.SequenceEqual for a way faster comparison. But Span.SequenceEqual is also very limited.

Reply:
This is not equals but "equivalent", basically null values are not taken into account
[1, 2, null] is equivalent to [1,2,3] but not to [10, 2, null]